app/serializers/profile_serializer.rb

```diff
@@ -6,7 +6,7 @@ class ProfileSerializer < ApplicationSerializer
   belongs_to :account
   belongs_to :benchmark
   belongs_to :parent_profile, record_type: :profile
-  has_many :rules
+  has_many :rules, &:rules_with_profile_ref_ids
```

Comment:
This solves the issue for a single profile. Within `/profiles?include=rules` this isn't solvable, as the structure is flat and objects are identified by uuid. Only the GraphQL has the profile-aware solved for a list of profiles.

```
profileA
  + rules
    + ruleAID
profileB
  + rules
    + ruleAID
included
  + rules
    + ruleA
```
`ruleA` cannot have `remediation_issue_id` for both profiles.

Reply:
In this case I'm only fixing this for GQL where it makes sense :see_no_evil: